zzhoritontalprogressbar/src/main/java/me/zhouzhuo/zzhoritontalprogressbar/ZzHorizontalProgressBar.java

```diff
@@ -0,0 +1,1182 @@
+
+        }
+    }
+
+    private void drawProgressCirclSecondProgressnoGradientlineShape(Canvas canvas, int width,
+                                                                    int secondProgressHeight, float secondPercent) {
+        //line shape
+        int secondProgressWidth = width - mPadding * 2 - secondProgressHeight;
+        float dx = secondProgressWidth * secondPercent;
+        Color hmosColor1 = ZzHorizontalProgressBar.changeParamToColor(mSecondProgressColor);
+        mSecondProgressPaint.setColor(hmosColor1);
+        //left circle
+        if (mSecondProgress == 0) {
+            if (mShowZeroPoint) {
+                canvas.drawCircle(mPadding
+                            + secondProgressHeight / 2.0f, mPadding
+                            + secondProgressHeight / 2.0f, secondProgressHeight / 2.0f, mSecondProgressPaint);
+            }
+        } else {
+            canvas.drawCircle(mPadding
+                        + secondProgressHeight / 2.0f, mPadding
+                        + secondProgressHeight / 2.0f, secondProgressHeight / 2.0f, mSecondProgressPaint);
+        }
+        //right circle
+        if (mSecondProgress == 0) {
+            if (mShowZeroPoint) {
+                canvas.drawCircle(mPadding
+                                + secondProgressHeight / 2.0f
+                                + dx, mPadding + secondProgressHeight / 2.0f,
+                            secondProgressHeight / 2.0f, mSecondProgressPaint);
+            }
+        } else {
+            canvas.drawCircle(mPadding + secondProgressHeight / 2.0f
+                                + dx, mPadding + secondProgressHeight / 2.0f,
+                        secondProgressHeight / 2.0f, mSecondProgressPaint);
+        }
+        //middle line
+        RectFloat rectF = new RectFloat(mPadding
+                    + secondProgressHeight / 2.0f, mPadding, mPadding
+                    + secondProgressHeight / 2.0f + dx, mPadding
+                    + secondProgressHeight);
+        canvas.drawRect(rectF, mSecondProgressPaint);
+
+    }
+
+    private void drawProgressCircleSecondProgressleftCircle(Canvas canvas, int secondProgressHeight) {
+        //left circle
+        if (mSecondProgress == 0) {
+            if (mShowZeroPoint) {
+                canvas.drawCircle(mPadding
+                        + secondProgressHeight / 2.0f, mPadding
+                        + secondProgressHeight / 2.0f, secondProgressHeight / 2.0f, mSecondGradientPaint);
+            }
+        } else {
+            canvas.drawCircle(mPadding
+                    + secondProgressHeight / 2.0f, mPadding
+                    + secondProgressHeight / 2.0f, secondProgressHeight / 2.0f, mSecondGradientPaint);
+        }
+
+    }
+
+    private void drawProgressCircleOpenGradientl(int width, float percent, int progressHeight) {
+        int progressWidth = width - mPadding * 2;
+        float dx;
```

Comment:
initialization and declaration ofthe variable can be in a single statement. (This applies to all the places wherever applicable, please check)

Reply:
Please make sure the distance between variable initialization & its first usage is not more than 5 lines. (CheckStyle issue)

Reply:
done